datahub-web-react/src/app/settings/AccessTokens.tsx

```diff
@@ -0,0 +1,200 @@
+                            <Select.Option key={duration.text} value={duration.duration}>
+                                {duration.text}
+                            </Select.Option>
+                        ))}
+                    </ExpirationDurationSelect>
+                    <Button
+                        disabled={!canGeneratePersonalAccessTokens}
+                        onClick={() => {
+                            generateAccessToken();
+                        }}
+                    >
+                        Generate Personal Access Token
+                    </Button>
+                </span>
+            </ExpirationSelectConainer>
+            {!canGeneratePersonalAccessTokens && (
+                <Typography.Paragraph type="secondary">
+                    Looks like you are not authorized to generate Personal Access Tokens. If you think this is
+                    incorrect, please contact your DataHub administrator.
+                </Typography.Paragraph>
+            )}
+
+            <Modal
+                width={700}
+                footer={null}
+                title={
+                    <Typography.Text>
+                        Generate a new <b>Personal Access Token</b>
+                    </Typography.Text>
+                }
+                visible={showModal}
+                onCancel={onClose}
+            >
+                <ModalSection>
+                    <ModalSectionHeader strong>Token</ModalSectionHeader>
+                    <ModalSectionParagraph>
+                        This token will expire in <b>{selectedExpiresInText}.</b>
+                    </ModalSectionParagraph>
+                    <Typography.Paragraph copyable={{ text: accessToken }}>
+                        <pre>{accessToken}</pre>
+                    </Typography.Paragraph>
+                </ModalSection>
+                <ModalSection>
+                    <ModalSectionHeader strong>Usage</ModalSectionHeader>
+                    <ModalSectionParagraph>
+                        To use the token, provide it as a <Typography.Text keyboard>Bearer</Typography.Text> token in
+                        the <Typography.Text keyboard>Authorization</Typography.Text> header when making API requests:
+                    </ModalSectionParagraph>
+                    <Typography.Paragraph copyable={{ text: accessTokenCurl }}>
+                        <pre>{accessTokenCurl}</pre>
+                    </Typography.Paragraph>
+                </ModalSection>
+                <ModalSection>
+                    <ModalSectionHeader strong>Learn More</ModalSectionHeader>
+                    <ModalSectionParagraph>
+                        To learn more about the DataHub APIs, check out the
+                        <a href="https://www.datahubproject.io/docs/"> DataHub Docs.</a>
+                    </ModalSectionParagraph>
+                </ModalSection>
+            </Modal>
+        </ContentContainer>
```

Comment:
it might be easier to read if this modal was factored out into its own component

Reply:
yes